### PR TITLE
Fix windows and cpp compatibility

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

https://github.com/rizinorg/cutter/pull/3008

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/include/rz_util/rz_time.h [hunk 1]
```diff
@@ -12,9 +12,14 @@
...
 #define ASCTIME_BUF_MINLEN 26
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #ifdef _MSC_VER
 struct timeval;
 
+#undef timezone
```
Comment (ret2libc) [inline librz/include/rz_util/rz_time.h]: Was this necessary in your tests? Otherwise, LGTM!

Comment (b1llow) [reply librz/include/rz_util/rz_time.h]: yes

Comment (ret2libc) [reply librz/include/rz_util/rz_time.h]: The reason I'm asking is that probably it is not too safe to undef a define defined by system header files.

Maybe we can rename it rz_timezone and use a 'define rz_timezone timezone' for the non-windows case. What do you think?

Comment (b1llow) [reply librz/include/rz_util/rz_time.h]: i think ok

**Review by ret2libc — APPROVED** [review]
LGTM. Did you already check it with Cutter?

**Comment by b1llow** [comment]
> LGTM. Did you already check it with Cutter?

YES

**Review by thestr4ng3r — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**